### ISSUE TITLE
[NBug] git checkout-index:  is not in the cache

### ISSUE THREAD

**Issue by huseyintahasalman** [gitextensions/gitextensions#11497]
### Environment

- Git Extensions 4.1.0.16698
- Build 1fe52a137acef1fbd2b653dea263b44ad5844177
- Git 2.42.0.windows.2
- Microsoft Windows NT 10.0.22631.0
- .NET 6.0.25
- DPI 96dpi (no scaling)

### Issue description

```
GitExtUtils.ExternalOperationException: git checkout-index:  is not in the cache ---> System.Exception: git checkout-index:  is not in the cache

   --- End of inner exception stack trace ---
   at GitCommands.ExecutableExtensions.GetOutputAsync(IExecutable executable, ArgumentString arguments, Byte[] input, Encoding outputEncoding, CommandCache cache, Boolean stripAnsiEscapeCodes) in C:\projects\gitextensions\GitCommands\Git\ExecutableExtensions.cs:line 128
   at Microsoft.VisualStudio.Threading.JoinableTask.CompleteOnCurrentThread()
   at Microsoft.VisualStudio.Threading.JoinableTask`1.CompleteOnCurrentThread()
   at Microsoft.VisualStudio.Threading.JoinableTaskFactory.Run[T](Func`1 asyncMethod, JoinableTaskCreationOptions creationOptions)
   at Microsoft.VisualStudio.Threading.JoinableTaskFactory.Run[T](Func`1 asyncMethod)
   at GitCommands.ExecutableExtensions.GetOutput(IExecutable executable, ArgumentString arguments, Byte[] input, Encoding outputEncoding, CommandCache cache, Boolean stripAnsiEscapeCodes) in C:\projects\gitextensions\GitCommands\Git\ExecutableExtensions.cs:line 42
   at GitCommands.GitModule.ResetFile(String file) in C:\projects\gitextensions\GitCommands\Git\GitModule.cs:line 1344
   at GitUI.GitUICommands.StartResetChangesDialog(String fileName) in C:\projects\gitextensions\GitUI\GitUICommands.cs:line 761
   at GitUI.GitUICommands.RunCommandBasedOnArgument(IReadOnlyList`1 args, IReadOnlyDictionary`2 arguments) in C:\projects\gitextensions\GitUI\GitUICommands.cs:line 1510
   at GitUI.GitUICommands.RunCommand(IReadOnlyList`1 args) in C:\projects\gitextensions\GitUI\GitUICommands.cs:line 1433
   at GitExtensions.Program.RunApplication() in C:\projects\gitextensions\GitExtensions\Program.cs:line 204
   at GitExtensions.Program.Main() in C:\projects\gitextensions\GitExtensions\Program.cs:line 105
```

no problem

### Steps to reproduce

a

### Did this work in previous version of GitExtensions?

a

### Diagnostics

a

**Comment by huseyintahasalman** [comment]
a

**Comment by gerhardol** [comment]
Duplicate of https://github.com/gitextensions/gitextensions/issues/10950
stage/unstage the file to remove conflict before reset

Fixed in 4.2.1
